src/ray/raylet/agent_manager.h

```diff
@@ -88,17 +89,36 @@ class AgentManager : public rpc::AgentManagerServiceHandler {
   virtual void DeleteRuntimeEnvIfPossible(const std::string &serialized_runtime_env,
                                           DeleteRuntimeEnvIfPossibleCallback callback);
 
+  /// Try to Get the information about the agent process,
+  /// if successful will return `ray::Status::OK`, otherwise will return
+  /// `ray::Status::Invalid`
+  ///
+  /// \param[out] agent_info The information of the agent process.
+  /// \return Status.
+  const ray::Status TryToGetAgentInfo(rpc::AgentInfo *agent_info) const {
+    if (IsAgentRegistered()) {
+      *agent_info = reported_agent_info_;
+      return ray::Status::OK();
+    } else {
+      std::string err_msg = "The agent has not finished register yet.";
+      return ray::Status::Invalid(err_msg);
+    }
+  }
+
  private:
   void StartAgent();
 
+  const bool IsAgentRegistered() const { return reported_agent_info_.id() == agent_id_; }
+
  private:
   Options options_;
-  pid_t reported_agent_id_ = 0;
-  int reported_agent_port_ = 0;
+  /// This is mainly to distinguish the initial value of `reported_agent_info_.id()`,
+  /// it is 0.
```

Comment:
I don't understand this comment?

Reply:
Also can you explain the difference between 0, -1, and other values here?

Reply:
I had modify it, please review it again.